### PR TITLE
fix: keycard signing integration into wallet connect flows

### PR DESCRIPTION
**Pull request by saledjenic**
Corresponding `nim-status-go` PR:
- https://github.com/status-im/nim-status-go/pull/32 

Fixes: #15957

These changes integrate keycard signing into each of so far the developed wallet connect flows.

In the video below you can see signing all of those flows using keycard:

https://github.com/user-attachments/assets/f74b08bd-6163-469d-affe-f7dbfe890c32

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/services/dapps/WalletConnectService.qml [hunk 1]
```diff
@@ -245,10 +245,6 @@ QObject {
 
         readonly property var supportedAccountsModel: SortFilterProxyModel {
             sourceModel: root.walletRootStore.nonWatchAccounts
-            filters: ValueFilter {
```
Comment (alexjba) [inline ui/app/AppLayouts/Wallet/services/dapps/WalletConnectService.qml:248]: 🥳

**Comment by anastasiyaig** [comment]
why ui tests are failing here constantly ... : (@saledjenic do you know?

**Review by alexjba — APPROVED** [review]
Awesome! Thank you! 🥳 

Fount 2 issues (not related to this PR). Will add separate tasks:
1. Incorrect authentication icon on `Sign` button when importing a keycard wallet account in a local profile
2. Cancelling authentication is not properly handled. We should reject the sign request or re-open the sign dialog.

**Comment by alexjba** [comment]
> why ui tests are failing here constantly ... : (@saledjenic do you know?

The authentication test needs to be updated.

The tests can be run directly from storybook:
1. `make -j10 run-storybook`
2. Go to `DAppsWorkflowPage`
3. Hit Run tests button in the footer

The output can be found in the console.
```
[2024-08-12T12:30:58.148Z] 3: QWARN  : QmlTests::DAppsRequestHandler::test_TestAuthentication() file:///home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-16064/ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml:95:5: QML Connections: Detected function "onSigningResult" in Connections element. This is probably intended to be a signal handler but no signal of the target matches the name.
[2024-08-12T12:30:58.148Z] 3: FAIL!  : QmlTests::DAppsRequestHandler::test_TestAuthentication() Compared values are not the same
[2024-08-12T12:30:58.148Z] 3:    Actual   (): password
[2024-08-12T12:30:58.148Z] 3:    Expected (): hello world
[2024-08-12T12:30:58.148Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-16064/storybook/qmlTests/tests/tst_DAppsWorkflow.qml(279)]
```

**Review by anastasiyaig — APPROVED** [review]

**Pull request merged**